frontend/web/src/app/components/material/material-list/material-list.component.ts

```diff
@@ -0,0 +1,170 @@
+import {AfterViewInit, Component, EventEmitter, OnInit, Output, ViewChild} from '@angular/core';
+import { Router } from '@angular/router';
+import { BsModalRef, BsModalService } from 'ngx-bootstrap';
+import {Material} from '../../../model/material';
+import {ErrorService} from '../../../services/error.service';
+import {MaterialService} from '../../../services/material/material.service';
+
+declare var $: any;
+
+@Component({
+  selector: 'app-material-list',
+  templateUrl: './material-list.component.html',
+  styleUrls: ['./material-list.component.scss'],
+  providers: [MaterialService, ErrorService],
+})
+export class MaterialListComponent implements OnInit, AfterViewInit {
+  @Output() selected: EventEmitter<Material> = new EventEmitter<Material>();
+  private table;
+  private materials: Material[];
+
+  private modalRef: BsModalRef;
+  @ViewChild('modalDelete') modalDelete;
+  private toBeDeleted: number = null;
+  selectedMaterial: Material = null;
+
+  constructor(private materialService: MaterialService,
+              private errorService: ErrorService,
+              private router: Router,
+              private modalService: BsModalService) {
+  }
+
+  ngOnInit() {
+    this.table = $('#table');
+  }
+
+  ngAfterViewInit(): void {
+    this.loadAndPopulate();
+  }
+
+  loadAndPopulate() {
+    /* get users */
+    this.materialService.getMaterials().subscribe(
+      (materials) => {
+        this.materials = materials;
+        this.populate();
+        this.prepareTriggers();
+      }, (error) => {
+        alert(error.verbose_message);
+      },
+    );
+  }
+
+  private populate() {
+    const _self = this;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assigning `this` reference to local variable not allowed: _self. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4327)